### PR TITLE
Bugfix: skip color cycler in Mosviz

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

The color cycler update in #1866 unintentionally changes the color of each spectrum visualized with Mosviz. Though the colors are fun, this isn't useful or intended. 

This PR only applies the color cycler for configs other than Mosviz. For Mosviz, only the default (gray) color is used. 

I've tested locally that Mosviz produces gray lines in the spectrum viewer, and e.g. Cubeviz still cycles colors. 

[🐱](https://jira.stsci.edu/browse/JDAT-2957)

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/mosviz/plugins/viewers.py [hunk 1]
```diff
@@ -358,6 +358,14 @@ def _on_row_selected(self, event):
                                     self._default_spectrum_viewer_reference_name,
                                     modified_prev_data, sender=self
                                 )
+                                # set back the counter by one in the spectrum viewer's
+                                # color cycler, so next spectrum added to the viewer has
+                                # same color as the previous one:
+                                spectrum_viewer = self.jdaviz_app.get_viewer(
+                                    self._default_spectrum_viewer_reference_name
+                                )
+                                spectrum_viewer.color_cycler.counter -= 1
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply jdaviz/configs/mosviz/plugins/viewers.py:367]: Good call – in testing I was able to switch between rows with models in such a way that the selected row's spectrum was shown in color other than gray. I'm about to push a commit that sets the `counter = -1` -- calls to the cycler always increment the counter by one, even on the zeroth call.

**Review by duytnguyendtn — APPROVED** [review]
Confirmed grey only in mosviz, but color cycles in specviz.

Agreed with Kyle, very elegant! It took me a second to read through the code and find where the solution was. Originally I was asking myself, "wait, where's the fix here?" 😅

**Comment by kecnry** [comment]
I'd vote for a squash here to get the other attempts out of the history on main... not sure if squash & merge plays nicely with the automatic backporter or if it needs to be squashed in advance 🤷

**Comment by rosteen** [comment]
> I'd vote for a squash here to get the other attempts out of the history on main... not sure if squash & merge plays nicely with the automatic backporter or if it needs to be squashed in advance 🤷

I seem to remember that they don't play well together, need to squash in advance.

**Pull request merged**